lib/web/auth/saml/index.js

```diff
@@ -16,7 +16,21 @@ passport.use(new SamlStrategy({
   callbackUrl: config.serverURL + '/auth/saml/callback',
   entryPoint: config.saml.idpSsoUrl,
   issuer: config.saml.issuer || config.serverURL,
-  cert: fs.readFileSync(config.saml.idpCert, 'utf-8'),
+  privateCert: config.saml.clientCert === undefined ? undefined : (function () {
+    try {
+      return fs.readFileSync(config.saml.clientCert, 'utf-8')
+    } catch (e) {
+      logger.error('saml client certificate not found at: ' + config.saml.clientCert)
```

Comment:
This is an almost silent error drop. We shouldn't do this. This would print `saml client certificate not found at: /path/top/cert.file` even when the certificate is in place but the permissions are wrong.

Reply:
Would it be okay to change the error message to something like "There was an error reading the client certificate at: /path/top/cert.file"/ Or should this have a more complex checking?

Reply:
I think you should also print `e.message`. Maybe something like

```js
logger.error(`SAML client certificate: ${e.message}`)
```

The logger already sets the log level to error, therefore we don't need to tell this in the error message. The prefix provides the needed pointer and the error message itself should provide the reason why things fail.